Update Set-ServiceNowAuth.ps1

GilmondRoss opened the pull request:
Fixes #2.

Rick-2CA commented:
I'll try to get all your PRs blessed and published as soon as I'm able.  Thanks for the contributions.

What do you think about changing the validation from looking for 'http...' to looking for regex pattern that covers more scenarios?  I only played with this a few minutes, but something like:

``` PowerShell
Param (
    [parameter(mandatory = $true)]
    [ValidateNotNullOrEmpty()]
    [ValidateScript({
        if ($_ -notmatch '^\w+\..*\.\w+') {
            Throw "The expected URL format is tenant.domain.com"
        }
    })]
    [string]$url
)
```

This seems to test properly for the following scenarios:
```
https://tenant.domain.com (False)
tenant.domain.com (True)
tenant.service-now.com (True)
domain.com (False)
tenant (False)
```

See what you think and choose whether you want to update the PR or not?  No matter the direction I'm likely to put this PR into dev and add some comment based help to fully satisfy the issue before publishing.

GilmondRoss commented:
Hi @Rick-2CA,

Initially I used regex, but thought I'd simplify it. I think your suggestion is safe enough by not assuming that the URL will **definitely** be tenant.service-now.com.

I am however getting varying error messages when testing your change.

https://tenant.domain.com returns this error, which looks fine and what I'd expect to see.
```
Set-ServiceNowAuth : Cannot validate argument on parameter 'url'. The expected URL format is tenant.domain.com
+ Set-ServiceNowAuth -url https://tenant.domain.com
```

But tenant.domain.com returns this error, which includes information about the regex. Not ideal as this could confuse the end user.
```
Set-ServiceNowAuth : Cannot validate argument on parameter 'url'. The " if ($_ -notmatch '^\w+\..*\.\w+') {
                    Throw "The expected URL format is tenant.domain.com"
                }
            " validation script for the argument with value "tenant.domain.com" did not return a result of True. Determine why the validation script failed, and then try the command again.
+ Set-ServiceNowAuth -url tenant.domain.com
```
This seemed to fix that.
```PowerShell
Param (
        [Parameter(Mandatory = $true)]
        [ValidateNotNullOrEmpty()]
        [ValidateScript( {
                if ($_ -match '^\w+\..*\.\w+') {
                    $true
                }
                else {
                    Throw "The expected URL format is tenant.domain.com"
                }
            })]
        [string]
        $Url,
```
I've also added comment-based help and changed some variable casing according to [best practice style guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#correct-format-for-hashtables-or-objects).

Let me know your thoughts.
